src/main/java/com/schedule/workout/workoutScheduler/services/WorkoutsService.java

```diff
@@ -22,19 +23,30 @@
     private IWorkoutsRepository workoutsRepository;
     @Autowired
     private IUsersRepository usersRepository;
+    @Autowired
+    IUserRoleRepository userRoleRepository;
+    @Autowired
+    IRolesRepository rolesRepository;
 
     //duration
     List<Integer> durationOfWorkout = asList(20, 30, 45, 60);
 
     //create  workout
     public CreateWorkoutModel createWorkout(CreateWorkoutModel createWorkoutModel) {
+    List<UserRoleDB> userRoles = userRoleRepository.findRolesByUserId(createWorkoutModel.getTrainerId());
+        userRoles.forEach(userRole -> {
+            String trainer = "Trainer";
+            if (!trainer.equals(userRole.getRoleDB().getName())) {
```

Comment:
I don't think this logic is good. Can you try to assign user 2 roles, Admin and Trainer and see if it works.